src/protocols/cosmic_workspaces/cosmic-workspaces.c

```diff
@@ -0,0 +1,675 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include <assert.h>
+#include <wlr/types/wlr_output.h>
+#include <wlr/util/log.h>
+#include "common/mem.h"
+#include "common/list.h"
+#include "cosmic-workspace-unstable-v1-protocol.h"
+#include "protocols/cosmic-workspaces.h"
+#include "protocols/cosmic-workspaces-internal.h"
+
+/*
+ *	.--------------------.
+ *	|        TODO        |
```

Comment:
Should this `TODO` still be here?

Reply:
Will work on the remaining items tomorrow and then remove the TODO but none of them should cause any issues.